code/__DEFINES/traits.dm

```diff
@@ -748,3 +748,20 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 
 /// This human wants to see the color of their glasses, for some reason
 #define TRAIT_SEE_GLASS_COLORS "see_glass_colors"
+
+// Radiation defines
+
+/// Marks that this object is irradiated
+#define TRAIT_IRRADIATED "iraddiated"
+
+/// Harmful radiation effects, the toxin damage and the burns, will not occur while this trait is active
+#define TRAIT_HALT_RADIATION_EFFECTS "halt_radiation_effects"
+
+/// This clothing protects the user from radiation.
+/// This should not be used on clothing_traits, but should be applied to the clothing itself.
+#define TRAIT_RADIATION_PROTECTED_CLOTHING "radiation_protected_clothing"
+
+/// Whether or not this item will allow the radiation SS to go through standard
+/// radiation processing as if this wasn't already irradiated.
+/// Basically, without this, COMSIG_PRE_POTENTIAL_IRRADIATION won't fire once the object is irradiated.
+#define TRAIT_BYPASS_EARLY_IRRADIATED_CHECK "radiation_bypass_early_irradiated_check"
```

Comment:
> > It's a SS method that takes an atom for an argument, but doesn't use src, so it can be safely converted into an atom method and redefined as needed.
> 
> What I meant is, redefine how in a way that suits _components_, in a way that is actually different to what I have now? Just firing a signal in there? Remember that the "geiger sound" component is used outside of the geiger counter, and should theoretically be applicable to any item/hardsuit/whatever.

My bad, you're right on that. I was thinking this was an innate trait held by a single type, always.
Didn't notice it was added by the component.

My suggestion then is to divide the check:
```dm
/atom/proc/can_be_irradiated()
	return FALSE

/obj/item/can_be_irradiated()
	if (HAS_TRAIT(target, TRAIT_RADIMMUNE))
		return FALSE
	return TRUE

/mob/living/carbon/human/can_be_irradiated()
	if (HAS_TRAIT(target, TRAIT_RADIMMUNE))
		return FALSE
	return TRUE
```

Then you change the `TRAIT_BYPASS_EARLY_IRRADIATED_CHECK` for something like `TRAIT_MINDFUL_OF_RADIATION` and substitute this:

`if (!can_irradiate_basic(target))`

For this:

`if (HAS_TRAIT(target, TRAIT_MINDFUL_OF_RADIATION) || (target.can_be_irradiated() && !HAS_TRAIT(target, TRAIT_IRRADIATED))`


That said, this exact implementation is just a suggestion.
Do it in any way you think is best, just make the names of the variables, procs and traits allude to the things we're simulating, more than to the specific implementation.
If you prefer to use `istype()` checks rather than redefined procs, that's fine as well, though I think the result is a little less flexible.

Must be noted that my suggested change does change the logic a little bit, in that it allows the `TRAIT_MINDFUL_OF_RADIATION` to be applied to anything, including things that are neither human nor items.

Reply:
>  If you prefer to use istype() checks rather than redefined procs, that's fine as well, though I think the result is a little less flexible.

It's less flexible, but the point of it was primarily performance--it's a lot cheaper to run native istypes than to call a proc on every atom, especially when this is being run over *everything* in range.

I'm gonna keep this for now but try to find another name, I definitely do think it's a hack but I'm not sure about the alternatives.